### PR TITLE
Fixed spelling mistakes in code and other files

### PR DESCRIPTION
**Pull request by bryancall**
Commands used:
```
codespell -w -L reenable,reenabled,anid,ned,larg,aci,tolen,aparent,toLen,ONL,doubleclick,ADMi,ded,ect,SEH
git co -- proxy/http2/hpack-tests
git co -- plugins/esi/test/parser_test.cc
git co -- doc
```

### CODE REVIEW

File: tests/include/catch.hpp [hunk 1]
```diff
@@ -10749,7 +10749,7 @@ namespace {
     }
 }
 
-#endif // signals/SEH handling
+#endif // signals/SHE handling
```
Comment (randall) [inline tests/include/catch.hpp:10752]: I think these should still be SEH (all in this file)

File: tests/gold_tests/pluginTest/regex_revalidate/regex_revalidate_miss.test.py [hunk 2]
```diff
@@ -164,7 +164,7 @@
 tr.StillRunningAfter = ts
 
 # 5 Stage - Change from MISS to STALE, reload
-tr = Test.AddTestRun("Reload config path1 STLE")
+tr = Test.AddTestRun("Reload config path1 STYLE")
```
Comment (randall) [inline tests/gold_tests/pluginTest/regex_revalidate/regex_revalidate_miss.test.py:167]: I think this is supposed to be STALE

File: tests/include/catch.hpp (hunk 1 above)
Comment (bryancall) [reply tests/include/catch.hpp:10752]: yep, I will see why it didn't get ignored - it was on my list

**Review by bneradt — APPROVED** [review]

**Review by randall — APPROVED** [review]

**Comment by zwoop** [comment]
This doesn't cherry-pick cleanly.

**Comment by zwoop** [comment]
Done via #8205 and #8206

**Pull request merged**